plugins/inputs/supervisor/supervisor_test.go

```diff
@@ -0,0 +1,124 @@
+package supervisor
+
+import (
+	"github.com/influxdata/telegraf/testutil"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func TestShort_SampleData(t *testing.T) {
```

Comment:
You might want to look into table-based tests. The starlark processor has a good [example](https://github.com/influxdata/telegraf/blob/c353bace3acb40946a80d21c4e479632534f06db/plugins/processors/starlark/starlark_test.go#L90). This makes the tests easily extensible.

Reply:
I'll try this later this week.

Reply:
Nicely done!